prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt

```diff
@@ -21,25 +21,25 @@ class HL7Reader(private val actionLogger: ActionLogger) : Logging {
     fun getMessages(rawMessage: String): List<Message> {
         if (rawMessage.isBlank()) {
             actionLogger.error(InvalidReportMessage("Provided raw data is empty."))
-            throw IllegalArgumentException("Empty Hl7 data")
         }
 
         val messages: MutableList<Message> = mutableListOf()
-        try {
-            val iterator = Hl7InputStreamMessageIterator(rawMessage.byteInputStream())
-            while (iterator.hasNext()) {
-                messages.add(iterator.next())
+        if (!actionLogger.hasErrors()) {
```

Comment:
Make this an else of the above if since this can only happen when the message is blank

Reply:
Updated